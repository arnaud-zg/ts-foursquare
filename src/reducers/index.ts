import { NVenues } from '../../types/venues'

export type TRootState = {
  venues: NVenues.IVenuesState
}
