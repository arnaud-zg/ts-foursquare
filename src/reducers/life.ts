import { createReducer } from 'typesafe-actions'
import { NLifeState } from '../../types'
import { putCredentials, TLifeAction } from '../actions'

export const initialState: NLifeState.IState = {
  status: false,
  credentials: {
    clientId: '',
    clientSecret: '',
  },
}

export const lifeReducer = createReducer<NLifeState.IState, TLifeAction>(
  initialState
).handleAction(
  putCredentials,
  (state, action): NLifeState.IState => ({
    ...state,
    status: true,
    credentials: {
      ...action.payload,
    },
  })
)
