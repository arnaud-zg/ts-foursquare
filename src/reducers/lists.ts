import { createReducer } from 'typesafe-actions'
import { NListsState } from '../../types'
import { getListsAsync, TListsAction } from '../actions'

export const initialState: NListsState.IState = {
  entities: {},
}

export const listsReducer = createReducer<NListsState.IState, TListsAction>(
  initialState
).handleAction(
  getListsAsync.success,
  (state, action): NListsState.IState => {
    return {
      ...state,
      entities: {
        ...state.entities,
        [action.payload.id]: action.payload,
      },
    }
  }
)
