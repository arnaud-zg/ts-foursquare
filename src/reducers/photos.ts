import { createReducer, getType } from 'typesafe-actions'
import { NPhotosState } from '../../types'
import { getPhotosDetailsAsync, TPhotosAction } from '../actions'

export const initialState: NPhotosState.IState = {
  entities: {},
}

export const photosReducer = createReducer<NPhotosState.IState, TPhotosAction>(
  initialState
).handleAction(
  getType(getPhotosDetailsAsync.success),
  (state, action): NPhotosState.IState => {
    return {
      ...state,
      entities: {
        ...state.entities,
        [action.payload.id]: action.payload,
      },
    }
  }
)
