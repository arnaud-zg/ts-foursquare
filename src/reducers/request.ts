import { createReducer, getType } from 'typesafe-actions'
import {
  NEntity,
  NLikes,
  NMiniVenue,
  NRecommendedPlaces,
  NRequestState,
  NVenue,
  NVenueListed,
  NVenuesCategories,
} from '../../types'
import { TRootAction } from '../actions'
import {
  getVenuesCategoriesAsync,
  getVenuesExploreAsync,
  getVenuesLikesAsync,
  getVenuesListedAsync,
  getVenuesNextVenuesAsync,
  getVenuesSearchAsync,
  getVenuesSimilarAsync,
  getVenuesSuggestCompletionAsync,
  getVenuesTrendingAsync,
} from '../actions/venues'
import { ASYNC_ACTION_NAME_MAPPING } from '../constants/asyncAction'
import { getVenuesListedGroupKey } from '../utils/venue'

export const initialState: NRequestState.IState = {}

export const requestReducer = createReducer<NRequestState.IState, TRootAction>(
  initialState
)
  .handleAction(
    getType(getVenuesCategoriesAsync.success),
    (state, action): NRequestState.IState => {
      const entityIds = action.payload.map(
        (item: NVenuesCategories.ICategory) => item.id
      )

      return {
        ...state,
        [ASYNC_ACTION_NAME_MAPPING[action.type]]: {
          entityIds,
        },
      }
    }
  )
  .handleAction(
    getType(getVenuesExploreAsync.success),
    (state, action): NRequestState.IState => {
      const entityIds = action.payload.map(
        (item: NRecommendedPlaces.IGroupItem) => item.venue.id
      )

      return {
        ...state,
        [ASYNC_ACTION_NAME_MAPPING[action.type]]: {
          entityIds,
        },
      }
    }
  )
  .handleAction(
    getType(getVenuesLikesAsync.success),
    (state, action): NRequestState.IState => {
      const entityIds = action.payload.items.map(
        (item: NLikes.IItem) => item.id
      )

      return {
        ...state,
        [ASYNC_ACTION_NAME_MAPPING[action.type]]: {
          entityIds,
        },
      }
    }
  )

  .handleAction(
    getType(getVenuesListedAsync.success),
    (state, action): NRequestState.IState => {
      const entityIds = action.payload.groups.map(
        (group: NEntity.IEntityGroup<NVenueListed.IGroupItem>) =>
          getVenuesListedGroupKey(group)
      )

      return {
        ...state,
        [ASYNC_ACTION_NAME_MAPPING[action.type]]: {
          entityIds,
        },
      }
    }
  )

  .handleAction(
    [
      getType(getVenuesNextVenuesAsync.success),
      getType(getVenuesSearchAsync.success),
      getType(getVenuesSimilarAsync.success),
      getType(getVenuesTrendingAsync.success),
    ],
    (state, action): NRequestState.IState => {
      const entityIds = action.payload.map((item: NVenue.IVenue) => item.id)

      return {
        ...state,
        [ASYNC_ACTION_NAME_MAPPING[action.type]]: {
          entityIds,
        },
      }
    }
  )
  .handleAction(
    getType(getVenuesSuggestCompletionAsync.success),
    (state, action): NRequestState.IState => {
      const entityIds = action.payload.map(
        (item: NMiniVenue.IMiniVenue) => item.id
      )

      return {
        ...state,
        [ASYNC_ACTION_NAME_MAPPING[action.type]]: {
          entityIds,
        },
      }
    }
  )
  .handleAction(
    [
      getType(getVenuesCategoriesAsync.cancel),
      getType(getVenuesExploreAsync.cancel),
      getType(getVenuesLikesAsync.cancel),
      getType(getVenuesNextVenuesAsync.cancel),
      getType(getVenuesSearchAsync.cancel),
      getType(getVenuesSimilarAsync.cancel),
      getType(getVenuesSuggestCompletionAsync.cancel),
      getType(getVenuesTrendingAsync.cancel),
      getType(getVenuesListedAsync.cancel),
    ],
    (state, action): NRequestState.IState => ({
      ...state,
      [ASYNC_ACTION_NAME_MAPPING[action.type]]: {
        cancelReason: action.payload,
        entityIds: [],
      },
    })
  )
  .handleAction(
    [
      getType(getVenuesCategoriesAsync.failure),
      getType(getVenuesExploreAsync.failure),
      getType(getVenuesLikesAsync.failure),
      getType(getVenuesNextVenuesAsync.failure),
      getType(getVenuesSearchAsync.failure),
      getType(getVenuesSimilarAsync.failure),
      getType(getVenuesSuggestCompletionAsync.failure),
      getType(getVenuesTrendingAsync.failure),
      getType(getVenuesListedAsync.failure),
    ],
    (state, action): NRequestState.IState => ({
      ...state,
      [ASYNC_ACTION_NAME_MAPPING[action.type]]: {
        entityIds: [],
        error: action.payload,
      },
    })
  )
