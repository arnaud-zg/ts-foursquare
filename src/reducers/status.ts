import { createReducer, getType } from 'typesafe-actions'
import { NStatus } from '../../types'
import {
  getListsAsync,
  getPhotosDetailsAsync,
  getVenuesCategoriesAsync,
  getVenuesExploreAsync,
  getVenuesLikesAsync,
  getVenuesListedAsync,
  getVenuesNextVenuesAsync,
  getVenuesSearchAsync,
  getVenuesSimilarAsync,
  getVenuesSuggestCompletionAsync,
  getVenuesTrendingAsync,
  TRootAction,
} from '../actions'
import { ASYNC_ACTION_NAME_MAPPING } from '../constants/asyncAction'

export const initialState: NStatus.IState = {}

export const statusReducer = createReducer<NStatus.IState, TRootAction>(
  initialState
)
  .handleAction(
    [
      getType(getListsAsync.request),
      getType(getPhotosDetailsAsync.request),
      getType(getVenuesCategoriesAsync.request),
      getType(getVenuesExploreAsync.request),
      getType(getVenuesLikesAsync.request),
      getType(getVenuesListedAsync.request),
      getType(getVenuesNextVenuesAsync.request),
      getType(getVenuesSearchAsync.request),
      getType(getVenuesSimilarAsync.request),
      getType(getVenuesSuggestCompletionAsync.request),
      getType(getVenuesTrendingAsync.request),
    ],
    (state, action: TRootAction): NStatus.IState => ({
      ...state,
      [ASYNC_ACTION_NAME_MAPPING[action.type]]: {
        ...state[action.type],
        hasError: false,
        isCancelled: false,
        isLoading: true,
      },
    })
  )
  .handleAction(
    [
      getType(getListsAsync.failure),
      getType(getPhotosDetailsAsync.failure),
      getType(getVenuesCategoriesAsync.failure),
      getType(getVenuesExploreAsync.failure),
      getType(getVenuesLikesAsync.failure),
      getType(getVenuesListedAsync.failure),
      getType(getVenuesNextVenuesAsync.failure),
      getType(getVenuesSearchAsync.failure),
      getType(getVenuesSimilarAsync.failure),
      getType(getVenuesSuggestCompletionAsync.failure),
      getType(getVenuesTrendingAsync.failure),
    ],
    (state, action: TRootAction): NStatus.IState => ({
      ...state,
      [ASYNC_ACTION_NAME_MAPPING[action.type]]: {
        ...state[action.type],
        hasError: true,
        isCancelled: false,
        isLoading: false,
      },
    })
  )
  .handleAction(
    [
      getType(getListsAsync.cancel),
      getType(getPhotosDetailsAsync.cancel),
      getType(getVenuesCategoriesAsync.cancel),
      getType(getVenuesExploreAsync.cancel),
      getType(getVenuesLikesAsync.cancel),
      getType(getVenuesListedAsync.cancel),
      getType(getVenuesNextVenuesAsync.cancel),
      getType(getVenuesSearchAsync.cancel),
      getType(getVenuesSimilarAsync.cancel),
      getType(getVenuesSuggestCompletionAsync.cancel),
      getType(getVenuesTrendingAsync.cancel),
    ],
    (state, action: TRootAction): NStatus.IState => ({
      ...state,
      [ASYNC_ACTION_NAME_MAPPING[action.type]]: {
        ...state[action.type],
        hasError: false,
        isCancelled: true,
        isLoading: false,
      },
    })
  )
  .handleAction(
    [
      getType(getListsAsync.success),
      getType(getPhotosDetailsAsync.success),
      getType(getVenuesCategoriesAsync.success),
      getType(getVenuesExploreAsync.success),
      getType(getVenuesLikesAsync.success),
      getType(getVenuesListedAsync.success),
      getType(getVenuesNextVenuesAsync.success),
      getType(getVenuesSearchAsync.success),
      getType(getVenuesSimilarAsync.success),
      getType(getVenuesSuggestCompletionAsync.success),
      getType(getVenuesTrendingAsync.success),
    ],
    (state, action: TRootAction): NStatus.IState => ({
      ...state,
      [ASYNC_ACTION_NAME_MAPPING[action.type]]: {
        ...state[action.type],
        hasError: false,
        isCancelled: false,
        isLoading: false,
      },
    })
  )
