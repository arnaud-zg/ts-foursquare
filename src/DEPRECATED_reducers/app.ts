import { combineReducers } from 'redux'
import { initialState as initialStateLife, lifeReducer } from './life'
import { initialState as initialStateLists, listsReducer } from './lists'
import { initialState as initialStatePhotos, photosReducer } from './photos'
import { initialState as initialStateRequest, requestReducer } from './request'
import { initialState as initialStateStatus, statusReducer } from './status'
import { initialState as initialStateVenues, venuesReducer } from './venues'

export const initialState = {
  life: initialStateLife,
  lists: initialStateLists,
  photos: initialStatePhotos,
  request: initialStateRequest,
  status: initialStateStatus,
  venues: initialStateVenues,
}

export type TState = typeof initialState

export const appReducer = combineReducers({
  life: lifeReducer,
  lists: listsReducer,
  photos: photosReducer,
  request: requestReducer,
  status: statusReducer,
  venues: venuesReducer,
})
