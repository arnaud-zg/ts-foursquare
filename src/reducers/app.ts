import { combineReducers } from 'redux'
import { initialState as initialStateLife } from '../../src/reducers/life'
import { initialState as initialStateStatus } from '../../src/reducers/status'
import { initialState as initialStateVenues } from '../../src/reducers/venues'
import { lifeReducer } from './life'
import { statusReducer } from './status'
import { venuesReducer } from './venues'

export const initialState = {
  life: initialStateLife,
  status: initialStateStatus,
  venues: initialStateVenues,
}

export type TState = typeof initialState

export const appReducer = combineReducers({
  life: lifeReducer,
  status: statusReducer,
  venues: venuesReducer,
})
