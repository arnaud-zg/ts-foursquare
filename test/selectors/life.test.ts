import { initialState } from '../../src/reducers/app'
import {
  lifeCredentialsSelector,
  lifeSelector,
  lifeStatusSelector,
  lifeAccessTokenSelector,
} from '../../src/selectors/life'
import { NStore } from '../../types'

const state: NStore.IState = {
  ...initialState,
  life: {
    status: true,
    credentials: {
      accessToken: '9123',
      clientId: '1234',
      clientSecret: '5678',
    },
  },
}

describe('selectors/life', () => {
  it('should get life state', () => {
    expect(lifeSelector(state)).toMatchSnapshot()
  })

  it('should get life status', () => {
    expect(lifeStatusSelector(state)).toMatchSnapshot()
  })

  it('should get lifeCredentialsSelector', () => {
    expect(lifeCredentialsSelector(state)).toMatchSnapshot()
  })

  it('should get lifeAccessTokenSelector', () => {
    expect(lifeAccessTokenSelector(state)).toMatchSnapshot()
  })
})
