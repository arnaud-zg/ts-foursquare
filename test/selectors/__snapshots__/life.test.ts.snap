// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectors/life should get life state 1`] = `
Object {
  "credentials": Object {
    "accessToken": "9123",
    "clientId": "1234",
    "clientSecret": "5678",
  },
  "status": true,
}
`;

exports[`selectors/life should get life status 1`] = `true`;

exports[`selectors/life should get lifeAccessTokenSelector 1`] = `"9123"`;

exports[`selectors/life should get lifeCredentialsSelector 1`] = `
Object {
  "accessToken": "9123",
  "clientId": "1234",
  "clientSecret": "5678",
}
`;
